### DISCUSSION TITLE
conda in non-default location and NOT listed among kernels

### DISCUSSION THREAD

**Discussion by thistlillo** [microsoft/vscode-jupyter#9366] [Questions and Answers] [answered]
On a remote linux machine I have installed conda in a non-default location. Neither the python microsoft extension nor the jupyter extension were able to find the python interpreter in the conda environment.

I modified the "Python: Conda path" (`python.condaPath`) variable in the remote settings to point to the `condabin` directory containing the conda executable. After this modification, the python extension detects the conda environments, that I can select using "Python: select interpreter". In [https://code.visualstudio.com/docs/python/environments](https://code.visualstudio.com/docs/python/environments) it  was suggested to modify `python.condaPath` in the user settings, but I am working remotely.

However, after modifying `python.condaPath`, the jupyter extension is not listing any of these environments, basically it seems to be unaffected by changes to the "Python: Conda path" variable.

Question: is there any way to have the jupyter extension to realize conda is not in its default location? 

Some info:
```
Version: 1.65.0 (user setup)
Commit: b5205cc8eb4fbaa726835538cd82372cc0222d43
Date: 2022-03-02T11:12:08.962Z
Electron: 13.5.2
Chromium: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Windows_NT x64 10.0.19043

Python extension by Microsoft: v2022.2.1924087327
Jupyter extension by Microsoft: v2022.2.1030672458
```

**Comment by DonJayamanne** [comment, marked as answer]
Thanks for this question. Please try re-loading VS Code, and the new conda path should be taken into account and conda environments should be listed in the Kernels list. Sometimes this can take a while as Python extension stores a cache of the conda environments and passes this (cached list) onto the Jupyter extension.

Please let me know if re-loading VS Code & waiting for a few seconds (some times 1-2 minutes) works or not.
If this doesn't work, please file an issue here and we'll take a look (please remember to provide the logs from the `Jupyter` output panel).

Also, please could you provide details on where the conda executable was located relative to the conda environments?
I.e. please could you provide the actual paths (feel free to obfuscate some of the sensitive information from the paths). If at all possible I'd like to ensure we locate such conda executables (based on the conda environments)

**Reply by thistlillo** [reply]
> Thanks for this question. Please try re-loading VS Code, and the new conda path should be taken into account and conda environments should be listed in the Kernels list. Sometimes this can take a while as Python extension stores a cache of the conda environments and passes this (cached list) onto the Jupyter extension.

Thank you for your support: I can confirm that the problem was solved with the relaunching of Visual Studio Code. Now the list of python interpreters in the Python extension and the list of the Python kernels in the Jupyter ext are the same.

> 
> Also, please could you provide details on where the conda executable was located relative to the conda environments? I.e. please could you provide the actual paths (feel free to obfuscate some of the sensitive information from the paths). If at all possible I'd like to ensure we locate such conda executables (based on the conda environments)

Both the conda application and the conda environments were placed on a mount point corresponding to persistent storage since the "local" disk is basically a Docker container (in a Kubernetes env) that gets destroyed every once in a while.
The full path of the conda home directory is:
```
/opt/project/conda/
```
The executable and the environments are located in the standard subfolders:
- `/opt/project/conda/bin/conda`
- `/opt/project/conda/envs`

Conda initialization code is in the standard place: `~/.bashrc`
